tokio-compat/src/runtime/threadpool/mod.rs

```diff
@@ -0,0 +1,405 @@
+/// use tokio_01::net::TcpListener;
+///
+/// # fn process<T>(_: T) -> Box<Future01<Item = (), Error = ()> + Send> {
+/// # unimplemented!();
+/// # }
+/// # fn dox() {
+/// # let addr = "127.0.0.1:8080".parse().unwrap();
+/// let listener = TcpListener::bind(&addr).unwrap();
+///
+/// let server = listener.incoming()
+///     .map_err(|e| println!("error = {:?}", e))
+///     .for_each(|socket| {
+///         tokio_01::spawn(process(socket))
+///     });
+///
+/// tokio_compat::run(server);
+/// # }
+/// # pub fn main() {}
+/// ```
+///
+/// # Panics
+///
+/// This function panics if called from the context of an executor.
+///
+/// [mod]: ../index.html
+pub fn run_03<F>(future: F)
+where
+    F: Future<Output = ()> + Send + 'static,
+{
+    // Check enter before creating a new Runtime...
+    let runtime = Runtime::new().expect("failed to start new Runtime");
+    runtime.spawn_03(future);
+    runtime.shutdown_on_idle();
+}
+
+impl Runtime {
+    /// Create a new runtime instance with default configuration values.
+    ///
+    /// This results in a reactor, thread pool, and timer being initialized. The
+    /// thread pool will not spawn any worker threads until it needs to, i.e.
+    /// tasks are scheduled to run.
+    ///
+    /// Most users will not need to call this function directly, instead they
+    /// will use [`tokio_compat::run`](fn.run.html).
+    ///
+    /// See [module level][mod] documentation for more details.
+    ///
+    /// # Examples
+    ///
+    /// Creating a new `Runtime` with default configuration values.
+    ///
+    /// ```
+    /// use tokio_compat::runtime::Runtime;
+    ///
+    /// let rt = Runtime::new()
+    ///     .unwrap();
+    ///
+    /// // Use the runtime...
+    /// ```
+    ///
+    /// [mod]: index.html
+    pub fn new() -> io::Result<Self> {
+        Builder::new().build()
+    }
+
+    /// Spawn a `futures` 0.1 future onto the Tokio runtime.
+    ///
+    /// This spawns the given future onto the runtime's executor, usually a
+    /// thread pool. The thread pool is then responsible for polling the future
+    /// until it completes.
+    ///
+    /// See [module level][mod] documentation for more details.
+    ///
+    /// [mod]: index.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio_compat::runtime::Runtime;
+    ///
+    /// fn main() {
+    ///    // Create the runtime
+    ///    let rt = Runtime::new().unwrap();
+    ///
+    ///    // Spawn a future onto the runtime
+    ///    rt.spawn(futures_01::future::lazy(|| {
+    ///        println!("now running on a worker thread");
+    ///        Ok(())
+    ///    }));
+    ///
+    ///    rt.shutdown_on_idle();
+    /// }
+    /// ```
+    ///
+    /// # Panics
+    ///
+    /// This function panics if the spawn fails. Failure occurs if the executor
+    /// is currently at capacity and is unable to spawn a new future.
+    pub fn spawn<F>(&self, future: F) -> &Self
```

Comment:
I expect _most_ code will spawn futures 0.1 and futures 0.3 futures by using the `tokio::spawn` free fns in the respective `tokio` main crates, but I thought we needed to provide a `Runtime::spawn` as well to ensure drop-in compatibility with the legacy runtime.

Again, I made `spawn` be the futures 0.1 version, so that legacy code still works with the compat runtime; `spawn_03` spawns futures 0.3 futures.

Reply:
> spawn_03 spawns futures 0.3 futures

Shouldn't that be `spawn_std`?

Reply:
As I mentioned in this comment https://github.com/tokio-rs/tokio/pull/1663#discussion_r335648613 I wasn't really sure what to name the std-future versions of these functions. `_std` might be better than `_03`. I was basing it off the precedent of the `futures` crate's `compat` module, which uses `01` and `03`, but I'm open to changing the naming scheme.

Reply:
At the very least, I don't think `03` makes sense; the critical attribute is the use of the std `Future` trait, right? The futures crate might update to 0.4 and beyond, but the std trait won't go anywhere.

Reply:
Changed the naming convention to `_std` in 30a020dd9fd72b977ca57ec318451bc84ff86dd9 :)